src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs

```diff
@@ -1230,6 +1240,8 @@ private static DkmClrCompilationResultFlags GetLocalResultFlags(LocalSymbol loca
                 var fieldType = field.Type;
                 var fieldKind = GeneratedNames.GetKind(field.Name);
                 if (fieldKind == GeneratedNameKind.DisplayClassLocalOrField ||
+                    fieldKind == GeneratedNameKind.TransparentIdentifier ||
+                    (fieldKind == GeneratedNameKind.AnonymousTypeField && IsTransparentIdentifierFieldInAnonymousType(field)) ||
```

Comment:
I actually left it separate for the parallelism, more than for the efficiency.  Changed.